components/VStreamfield.vue

```diff
@@ -4,76 +4,77 @@ const props = defineProps<{
   streamfieldBlocks: StreamfieldBlock[]
 }>()
 const emit = defineEmits<{
-  (e: 'allBlocksMounted'): void
+  (e: 'allBlocksMounted', value: any): void
 }>()
+const el = ref(null)
 
 let blocksMounted = 0
 
 function countMountedBlock() {
   blocksMounted++
   if (blocksMounted === props.streamfieldBlocks.length) {
-    emit('allBlocksMounted')
+    emit('allBlocksMounted', el)
   }
 }
 </script>
 
 <template>
-  <div class="streamfield">
+  <div class="streamfield" ref="el">
     <template v-for="block in streamfieldBlocks">
       <StreamfieldBlockQuote
         v-if="block.type === 'block_quote'"
         :key="`${block.id}-block-quote`"
         :block="block"
-        @hook:mounted="countMountedBlock"
+        @vnodeMounted="countMountedBlock"
       />
 
       <StreamfieldCode
         v-else-if="block.type === 'code'"
         :key="`${block.id}-code`"
         :block="block"
-        @hook:mounted="countMountedBlock"
+        @vnodeMounted="countMountedBlock"
       />
 
       <StreamfieldContentCollection
         v-else-if="block.type === 'content_collection'"
         :key="`${block.id}-content-collection`"
         :block="block"
-        @hook:mounted="countMountedBlock"
+        @vnodeMounted="countMountedBlock"
       />
 
       <StreamfieldEmbed
         v-else-if="block.type === 'embed'"
         :key="`${block.id}-embed`"
         :block="block"
-        @hook:mounted="countMountedBlock"
+        @vnodeMounted="countMountedBlock"
       />
 
       <StreamfieldHeading
         v-else-if="block.type === 'heading'"
         :key="`${block.id}-heading`"
         :block="block"
-        @hook:mounted="countMountedBlock"
+        @vnodeMounted="countMountedBlock"
       />
 
       <StreamfieldImage
         v-else-if="block.type === 'image'"
         :key="`${block.id}-image`"
         :block="block"
-        @hook:mounted="countMountedBlock"
+        @vnodeMounted="countMountedBlock"
       />
 
       <StreamfieldParagraph
         v-else-if="block.type === 'paragraph'"
         :key="`${block.id}-paragraph`"
         :block="block"
-        @hook:mounted="countMountedBlock"
+        @vnodeMounted="countMountedBlock"
       />
 
       <StreamfieldPullQuote
         v-else-if="block.type === 'pull_quote'"
         :key="`${block.id}-pull-quote`"
         :block="block"
-        @hook:mounted="countMountedBlock"
+        @vnodeMounted="countMountedBlock"
```

Comment:
didn't know about vnodeMounted - is hook:mounted deprecated?

Reply:
Not sure where I saw this one but it works. Looks like I should actually maybe be using `vue:mounted` https://v3-migration.vuejs.org/breaking-changes/vnode-lifecycle-events.html#_3-x-syntax